### PR TITLE
BAEL-7470

### PR DESCRIPTION
**Pull request by shahulbasha**
BAEL-7470 How to Autowire a Spring Bean in a Servlet Filter

### CODE REVIEW

File: spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterAutowiringSupport.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        SpringBeanAutowiringSupport.processInjectionBasedOnServletContext(this,
+            filterConfig.getServletContext());
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpServletRequest=(HttpServletRequest)servletRequest;
```
Comment (KevinGilmore) [inline spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterAutowiringSupport.java:31]:
```suggestion
        HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
```

File: spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterAutowiringSupport.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        loggingService.log(httpServletRequest.getMethod(),httpServletRequest.getRequestURI());
```
Comment (KevinGilmore) [inline spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterAutowiringSupport.java:32]:
```suggestion
        loggingService.log(httpServletRequest.getMethod(), httpServletRequest.getRequestURI());
```

File: spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterAutowiringSupport.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        filterChain.doFilter(servletRequest,servletResponse);
```
Comment (KevinGilmore) [inline spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterAutowiringSupport.java:33]:
```suggestion
        filterChain.doFilter(servletRequest, servletResponse);
```

File: spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterDelegatingFilterProxy.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+public class LoggingFilterDelegatingFilterProxy implements Filter {
+
+    @Autowired
+    LoggingService loggingService;
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpServletRequest=(HttpServletRequest)servletRequest;
+        loggingService.log(httpServletRequest.getMethod(),httpServletRequest.getRequestURI());
+        filterChain.doFilter(servletRequest,servletResponse);
```
Comment (KevinGilmore) [inline spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterDelegatingFilterProxy.java:25]: Add spacing as was suggested in the previous file.

File: spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterWebApplicationContext.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        loggingService = WebApplicationContextUtils
+            .getRequiredWebApplicationContext(filterConfig.getServletContext())
+            .getBean(LoggingService.class);
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpServletRequest=(HttpServletRequest)servletRequest;
+        loggingService.log(httpServletRequest.getMethod(),httpServletRequest.getRequestURI());
+        filterChain.doFilter(servletRequest,servletResponse);
```
Comment (KevinGilmore) [inline spring-core-4/src/main/java/com/baeldung/filter/LoggingFilterWebApplicationContext.java:32]: spacing

File: spring-core-4/src/main/java/com/baeldung/filter/UserController.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserController {
+
+    @GetMapping("/users")
+    public List<User> getUsers(){
+        return Arrays.asList(new User("1","John","john@email.com")
+            ,new User("2","Smith","smith@email.com"));
```
Comment (KevinGilmore) [inline spring-core-4/src/main/java/com/baeldung/filter/UserController.java:15]: move comma to end of previous line

**Review by KevinGilmore — COMMENTED** [review]
Add whitespace as noted, and move comma as noted.

**Comment by shahulbasha** [comment]
moving this p.r to new module

**Pull request closed without merging**